src/check-cluster.ts

```diff
@@ -6,11 +7,19 @@
 
 import { knExecutor } from './cli/execute';
 import { KubectlAPI } from './cli/kubectl-api';
+import { telemetryLog } from './telemetry';
 
-export async function checkOpenShiftCluster(): Promise<boolean> {
+export async function checkOpenShiftCluster(checkClusterVersion?: boolean): Promise<boolean> {
   try {
     const result = await knExecutor.execute(KubectlAPI.checkOcpCluster(), process.cwd(), false);
     if (result?.stdout?.trim()) {
+      if (checkClusterVersion) {
+        telemetryLog(
```

Comment:
Why sending ocp version only if user performs a deploy? Wouldn't it be better to send it during activation or cluster switch? `The user is connected to kubernetes or ocp version blabla`? Then you know all following actions are performed to it. If the user executes 2-3 deploys the ocp version would always be the same. Why sending it each time?

Reply:
done